dq-server/src/controllers/fileImporter.js

```diff
@@ -0,0 +1,102 @@
+const { Question } = require('../db');
+const _ = require('lodash');
+const csv = require('csv-parser');
+const fs = require('fs');
+const { getThemes } = require('./themes');
+const { createQuestion, getQuestionByText } = require('./questions');
+const { createQuestionSchema } = require('../validation/input');
+
+async function importQuestions(stream) {
+	const streamData = await readCSVStream(stream, verifyCSVLine, () => {});
+	streamData.errors = filterNullElements(streamData.errors);
+	if (!_.isEmpty(streamData.errors)) return streamData;
+	let result = [];
+	for await (const question of streamData.questionsToAdd) {
+		const questionAdded = await createQuestion(question);
+		result.push(questionAdded);
+	}
+	return streamData;
+}
+
+
+async function readCSVStream(stream, callback, callbackOnEnd){
+	const themeSet = await createThemeSet();
```

Comment:
I think you can have a callbackOnSuccess and a callbackOnError. You can call the error at the moment you catch one and stop the method.

Then the line 12 and below will be called in the OnSuccess and you don't need to check if an array is empty or not